Assets/Scripts/PlayerController.cs

```diff
@@ -30,20 +31,23 @@ public class PlayerController : MonoBehaviour
 
     public int GetCurrentHealth() => currentHealth;
 
+    public void SetCanPlayerMove(bool boo) { canPlayerMove = boo; }
+
     private void Awake()
     {
         rb2D = GetComponent<Rigidbody2D>();
         targetPosition = rb2D.position;
         currentHealth = maxHealth;
         playerColl = this.GetComponent<Collider2D>();
         nextFire = Time.time;
+        canPlayerMove = true;
     }
 
     private void Update()
     {
         if (PauseController.Instance.GamePaused) {  return;  }
 
-        if (Input.touchCount > 0)
+        if (canPlayerMove && Input.touchCount > 0)
```

Comment:
The player can still move while playing a card if they tap a destination position before playing a card. I think the player should stop moving immediately when a card starts being played, but whether to do that is subjective. To implement that behavior, `targetPosition` could be set to `rb2D.position` when `canPlayerMove` is set to false.